### ISSUE TITLE
Items Groups broken

### ISSUE THREAD

**Issue by thecyrus** [TheGameCreators/GameGuruRepo#5158]
I found out that a lot of asset in a level of mine are grouped on, and I can't edit the group nor ungroup them, no matter if I try to edit or ungroup them.
At times they get free, I don't think I did something wrong about it.
Did anyone notice the same thing?

**Comment by imothep85** [comment]
i have the same problems, if you create for example in my case a cave crystal and you add on top of that crystal a lighting, and group the light and the crystal, if you rotate the object the light doesn't follow, rotate with the object, the grouped light stay in place when you rotate the object....

**Comment by LeeBamberTGC** [comment]
@thecyrus This can happen with older levels that have some corruption in the group lists. Does it happen with a fresh new level created with the latest build, and if so, can you attach one here using stock media and give me a step by step what I should do with the level to see the issue, thanks!

**Comment by imothep85** [comment]
it happen with fresh new contents... i started from zero, flat level, added my props, and that problem is very old now.

**Comment by LeeBamberTGC** [comment]
@imothep85 can you attach one here using stock media and give me a step by step what I should do with the level to see the issue, thanks!

**Comment by imothep85** [comment]
well what i created is i imported my asset a crystal and i link color lights to it, 
i mean i group them, the problem is when i rotate the group, ONLY the mesh rotate, but not the lights... 
![ggmax_cave_rocks48](https://github.com/TheGameCreators/GameGuruRepo/assets/8946215/ca44392b-6312-46d4-8aff-ace68466832c)

**Comment by LeeBamberTGC** [comment]
@imothep85 Can you reproduce this issue with some stock asset and lights, and then attach your small test level here, thanks!

**Comment by imothep85** [comment]
i tested it right now with stock medias and my props and everything works strange...

**Comment by imothep85** [comment]
i mean it works :D,

**Comment by mike123yy** [comment]
> @thecyrus This can happen with older levels that have some corruption in the group lists. Does it happen with a fresh new level created with the latest build, and if so, can you attach one here using stock media and give me a step by step what I should do with the level to see the issue, thanks!

Is there any way to fix the older levels with this problem? I have my game I built that I was updating and making a couple more new levels and when I went back to the older levels several objects are now grouped which I never grouped and there's no way to ungroup them or even get rid of them. It wouldn't really matter if the item still worked but it's grouped some sound zones to objects and now the sounds won't work.

**Comment by LeeBamberTGC** [comment]
@mike123yy Can you attach a shot of what is in the Group List for the older corrupted level, thanks:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/bd3bc9cb-720e-4671-a2e9-cace43e01c2d)

**Comment by mike123yy** [comment]
![3](https://github.com/TheGameCreators/GameGuruRepo/assets/41445779/64241078-55c1-44fb-ab1c-8303cc18568a)
![bug](https://github.com/TheGameCreators/GameGuruRepo/assets/41445779/1f2e512c-eeb5-4950-abb6-713bf07eaf62)


Here's a screenshot of the list and another screenshot of a sound zone attached to some fire pits. I could only get one fire pit in the screenshot though at that angle. Back when I made this level I never even use groups so they seem to have made themselves on their own. Most of the objects that are grouped are quite a ways away from each other. I really couldn't care if they're grouped or not but with the audio zones it seems to have affected the way they function. On my other level I decided to just fix some other stuff while waiting for a reply and for some reason the audio zone became unattached from the group so I was able to delete it and make a new one which fix that problem. Maybe it fix something by saving after my other fixes? I was going to work on this level today and maybe I can glitch it into ungrouping itself enough to get it fixed. I did notice though with all these groups that made themselves they won't allow you to ungroup them.

**Comment by LeeBamberTGC** [comment]
@mike123yy There is a self repair system in the newer versions that can inspect group data and delete any that make no sense, i.e. two groups that use the same object for example. This cleanup may have been responsible for your other level that repaired itself somewhat.  For the level you are struggling with, looks like you have four internal groups doing on:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/adcf138c-eb83-407f-a120-d3e0f7679d3a)

One way to remove the group is to select any object from the Group List that relates to the group, then move the mouse pointer to the main level editing area, then press DELETE key that will delete the group and its objects, then use CTRL+Z to undo the object deletes. It will restore your objects and zone, but the group is not restored which is what you will want.

**Comment by mike123yy** [comment]
If the grouping was my only problem it would definitely be fixed. I looked back at all the levels and they no longer have any funny groups. All the other bugs I sent you in the first message are still there though. I've been working at it all day trying different methods and I can get the problems fixed when testing the game but they still keep breaking when I build a standalone. The final level with issues was built within the last month so it's issues wouldn't be from it being an old level. Also for some reason my custom character that uses the NPC script for control will no longer drop loot in the standalone or the test level now. I also tested the save system with checkpoints after rebuilding it and they still cause quite a few bugs when loading saves as well. I'm kind of at a loss at how to fix any of the problems

    On Monday, July 8, 2024 at 05:45:34 AM CDT, Lee Bamber ***@***.***> wrote:  
 
 


@mike123yy There is a self repair system in the newer versions that can inspect group data and delete any that make no sense, i.e. two groups that use the same object for example. This cleanup may have been responsible for your other level that repaired itself somewhat. For the level you are struggling with, looks like you have four internal groups doing on:
image.png (view on web)

One way to remove the group is to select any object from the Group List that relates to the group, then move the mouse pointer to the main level editing area, then press DELETE key that will delete the group and its objects, then use CTRL+Z to undo the object deletes. It will restore your objects and zone, but the group is not restored which is what you will want.
